Sources/SwiftDriver/Utilities/DateAdditions.swift

```diff
@@ -10,27 +10,132 @@
+///             products that are expected to transit between machines.
+public struct TimePoint: Equatable, Comparable, Hashable {
+  public var seconds: UInt64
+  public var nanoseconds: UInt32
+
+  public init(seconds: UInt64, nanoseconds: UInt32) {
+    self.seconds = seconds
+    self.nanoseconds = nanoseconds
+  }
+
+  public static func < (lhs: TimePoint, rhs: TimePoint) -> Bool {
+    return (lhs.seconds, lhs.nanoseconds) < (rhs.seconds, rhs.nanoseconds)
+  }
+}
+
+extension TimePoint {
+  public static func seconds(_ value: Int) -> TimePoint {
+    precondition(value >= 0,
+                 "Duration value in seconds is \(value), but cannot be negative")
+    return TimePoint(seconds: UInt64(value), nanoseconds: 0)
+  }
+
+  public static func nanoseconds(_ value: Int) -> TimePoint {
+    precondition(value >= 0,
+                 "Duration value in nanoseconds is \(value), but cannot be negative")
+    let (seconds, nanos) = value.quotientAndRemainder(dividingBy: TimePoint.nanosecondsPerSecond)
+    return TimePoint(seconds: UInt64(seconds),
+                     nanoseconds: UInt32(nanos))
+  }
+}
+
+extension TimePoint: AdditiveArithmetic {
+  public static var zero: TimePoint {
+    return .seconds(0)
+  }
+
+  public static func + (lhs: TimePoint, rhs: TimePoint) -> TimePoint {
+    // Add raw operands
+    var seconds = lhs.seconds + rhs.seconds
+    var nanos = lhs.nanoseconds + rhs.nanoseconds
+    // Normalize nanoseconds
+    if nanos >= TimePoint.nanosecondsPerSecond {
+      nanos -= UInt32(TimePoint.nanosecondsPerSecond)
+      seconds += 1
+    }
+    return TimePoint(seconds: seconds,
+                    nanoseconds: nanos)
   }
-  guard secsAndNanos.count == 2 else {
-    throw Errors.needSecondsAndNanoseconds
+
+  public static func - (lhs: TimePoint, rhs: TimePoint) -> TimePoint {
+    // Subtract raw operands
+    var seconds = lhs.seconds - rhs.seconds
+    // Normalize nanoseconds
+    let nanos: UInt32
+    if lhs.nanoseconds >= rhs.nanoseconds {
+      nanos = lhs.nanoseconds - rhs.nanoseconds
+    } else {
+      // Subtract nanoseconds with carry - order of operations here
+      // is important to avoid overflow.
+      nanos = lhs.nanoseconds + UInt32(TimePoint.nanosecondsPerSecond) - rhs.nanoseconds
+      seconds -= 1
     }
-    self = Self(legacyDriverSecs: secsAndNanos[0], nanos: secsAndNanos[1])
+    return TimePoint(seconds: seconds,
+                     nanoseconds: nanos)
+  }
+}
+
+extension TimePoint{
+  public static func now() -> TimePoint {
+    #if os(Windows)
+    var stTime: SYSTEMTIME = SYSTEMTIME()
+    var ftTime: FILETIME = FILETIME()
+
+    GetSystemTime(&stTime)
+    SystemTimeToFileTime(&stTime, &ftTime)
+
+    let result: UInt64 = (UInt64(ftTime.dwLowDateTime) << 0)
+                       + (UInt64(ftTime.dwHighDateTime) << 32)
+    // Windows ticks in 100 nanosecond intervals.
+    return .seconds(result / 10_000_000)
```

Comment:
This seems to need an explicit conversion ...
```suggestion
    return .seconds(Int(result / UInt64(10_000_000)))
```